src/adapters/scene-fetcher.ts

```diff
@@ -194,14 +202,26 @@ export async function createSceneFetcherComponent(
     return namesResponse.elements.some((element) => element.name.toLowerCase() === nameToValidate)
   }
 
+  async function hasWorldStreamingPermission(authAddress: string, worldName: string): Promise<boolean> {
+    const permissionsOverWorld = await fetchOnWorldActionPermissions(worldName)
+    let hasStreamingPermission = false
+    if (permissionsOverWorld?.streaming.type === PermissionType.AllowList) {
+      hasStreamingPermission = permissionsOverWorld.streaming.wallets.includes(authAddress)
+    }
+
+    return hasStreamingPermission
+  }
+
   return {
     fetchWorldPermissions,
     fetchScenePermissions,
+    fetchOnWorldActionPermissions,
     getPlaceByParcel,
     getWorldByName,
     getPlace,
     getAddressResources,
     hasLandPermission,
-    hasWorldPermission
+    hasWorldOwnerPermission,
+    hasWorldStreamingPermission
   }
```

Comment:
WDYT about creating two new adapters, `place-fetcher` and `world-fetcher`, to help us retrieve resources from the right services. This way, the scene-fetcher can focus on figuring out which adapter to use for each situation.

I'd keep the current `scene-fetcher` signature, so the refactor is not huge.